proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java

```diff
@@ -105,6 +123,47 @@ private static List<ReportPoint> fromOtlpRequest(
     return wfPoints;
   }
 
+  @VisibleForTesting
+  static List<KeyValue> replaceServiceNameKeyWithServiceKey(List<KeyValue> attributes) {
+    Optional<KeyValue> serviceNameAttr =
+        attributes.stream()
+            .filter(kv -> kv.getKey().equals(OtlpTraceUtils.OTEL_SERVICE_NAME_KEY))
+            .findFirst();
+
+    if (serviceNameAttr.isPresent() && !isServiceKeyPresent(attributes)) {
```

Comment:
What do you think of getting rid of the `isServiceKeyPresent()` method, and re-using `OtlpTraceUtils.getAttrByKey(attributes, SERVICE_TAG_KEY)`?

The conditional would become: `if (serviceNameAttr.isPresent() && !serviceAttr.isPresent()) {`

Reply:
makes sense, will do